yb-voyager/cmd/live_migration.go

```diff
@@ -16,42 +16,65 @@ limitations under the License.
 package cmd
 
 import (
-	"encoding/json"
+	"errors"
 	"fmt"
 	"os"
 	"path/filepath"
+	"time"
 
 	log "github.com/sirupsen/logrus"
 
 	"github.com/yugabyte/yb-voyager/yb-voyager/src/tgtdb"
-	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
 )
 
 func streamChanges() error {
-	queueFilePath := filepath.Join(exportDir, "data", "queue.json")
-	log.Infof("Streaming changes from %s", queueFilePath)
-	file, err := os.OpenFile(queueFilePath, os.O_CREATE, 0640)
-	if err != nil {
-		return fmt.Errorf("error opening file %s: %v", queueFilePath, err)
+	eventQueue := tgtdb.NewEventQueue(exportDir)
+	log.Infof("Streaming changes from %s", eventQueue.QueueDirPath)
+	for { // continuously get next segments to stream
+		segment, err := eventQueue.GetNextSegment()
+		log.Infof("Got next segment to stream: %v", segment)
+		if err != nil {
+			if segment == nil && errors.Is(err, os.ErrNotExist) {
```

Comment:
Check if we can incorporate a max retry count. We don't want to be trying forever, silently.

Reply:
Check this - https://pkg.go.dev/github.com/cenkalti/backoff/v4#example-Retry

Reply:
are you suggesting to exit after max retries? I think we want to keep the importer running forever.

I was thinking of exponential backoff(with some max sleep time) but without exiting, but that won't make much difference i think rather we can keep a larger value for sleep here

Reply:
Hmm, I guess you're right. It's tricky. If exporter has a problem and shuts down before creating the new segment, we want the importer to keep waiting, not shutdown.